File: app/src/main/java/com/lxk/animandview/BaseClickActivity.java
package com.lxk.animandview;

import android.view.View;

import androidx.annotation.IdRes;
import androidx.appcompat.app.AppCompatActivity;

/**
 * @author https://github.com/103style
 * @date 2019/12/11 14:08
 */
public abstract class BaseClickActivity extends AppCompatActivity implements View.OnClickListener {

    protected void setClickListener(@IdRes int... ids) {
        for (int id : ids) {
            findViewById(id).setOnClickListener(this);
        }
    }
}

File: app/src/main/java/com/lxk/animandcustomviewdemo/BaseClickActivity.kt
package com.lxk.animandcustomviewdemo

import android.view.View

import androidx.annotation.IdRes
import androidx.appcompat.app.AppCompatActivity

/**
 * @author https://github.com/103style
 * @date 2020/1/10 9:50
 */
abstract class BaseClickActivity : AppCompatActivity(), View.OnClickListener {
    protected fun setClickListener(@IdRes vararg ids: Int) {
        for (id in ids) {
            findViewById<View>(id).setOnClickListener(this)
        }
    }
}